### PR TITLE
fix(ci): guard gateway daily-log async destination against unhandled flush errors

### PR DESCRIPTION
**Pull request by asafgolombek**
## What

Hardens `createGatewayPinoLogger` in `packages/gateway/src/platform/gateway-log-file.ts`:

- `mkdirSync(logDir, { recursive: true })` before constructing the destination, so a missing directory can never throw out of logger construction.
- Attaches a swallowing `error` listener to the `sync: false` SonicBoom destination. The daily log is best-effort, so async open/flush failures must not escape as unhandled `error` events.

## Why

The gateway daily-log uses `pino.destination({ sync: false })`, which opens and flushes via a background SonicBoom stream on a later tick. With no `error` listener, an async open/flush failure becomes an **unhandled `error` event** that can crash the process or be mis-attributed to an unrelated in-flight test.

This surfaced as a **macOS-only flake** on the cross-platform CI leg (observed on PR #614's release job, run 27494059073): a short-lived logger built in one test would async-open its log file *after* that test's temp dir had already been `rmSync`'d, emitting an `ENOENT` that failed the next test — `gateway-log-file.test.ts > createGatewayPinoLogger > appends JSON to daily log file when stdout is not a TTY`:

```
ENOENT: no such file or directory, open '/var/folders/.../T/nimbus-gw-log-level-.../gateway-2026-06-14.log'
  syscall: "open", errno: -2, code: "ENOENT"
(fail) ... appends JSON to daily log file when stdout is not a TTY [2.23ms]
```

The ~2ms failure time (well under the test's 5000ms poll deadline) confirms it was the unhandled async-stream error, not the poll helper timing out.

## Verification

- Scoped test green **5/5 runs** (was non-deterministic): `bun test packages/gateway/src/platform/gateway-log-file.test.ts` → 32 pass / 0 fail.
- Full `bun run preflight`: green on all gates. `audit:coverage-floor` failed only on the local **Windows** lcov (known Windows≠Linux skew); verified authoritatively on Docker-Linux (`oven/bun:latest`, bun 1.3.14) — **coverage-floor PASS**. The changed file is at **90.9% line / 88.5% branch** on Linux; the new `error`-handler arrow is covered by existing tests.
- `tsc --noEmit` (gateway): clean. `biome check`: clean.

## Notes

This does **not** itself unblock PR #614 (that's the release-please bot branch and just needs its flaky macOS job re-run); this is the durable repo-wide remedy so the flake stops recurring.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Enhanced logging resilience to gracefully handle directory creation failures and async log-writing errors, preventing application crashes.
  * Logger now maintains best-effort logging even when filesystem issues occur.

* **Tests**
  * Added test coverage for logger error handling in edge-case scenarios.

### CODE REVIEW

**Comment by asafgolombek** [comment]
Addressed in d01f6240: wrapped `mkdirSync` in try/catch (mirroring `emergencyGatewayLog`) so directory-creation failures (EACCES/ENOSPC/EROFS) are swallowed and the async destination `error` handler absorbs the subsequent open failure — the best-effort guarantee now holds. Added a test that forces an ENOTDIR mkdir failure and asserts the logger does not throw, covering the new catch arm.

File: .claude/settings.local.json [hunk 1]
```diff
@@ -19,7 +19,10 @@
       "Bash(gh pr *)",
       "Bash(bun -e ' *)",
       "Skill(superpowers:using-git-worktrees)",
-      "PowerShell(bun run *)"
+      "PowerShell(bun run *)",
+      "Read(//tmp/**)",
+      "Bash(break)",
+      "Bash(:)"
```
[the remark this answers is not in the document]
Comment (asafgolombek) [reply .claude/settings.local.json:25]: This is stale — `.claude/settings.local.json` was modified only in an earlier commit (2c739cc2) that accidentally swept in a regenerated artifact. That commit was amended out; the current PR diff touches only `packages/gateway/src/platform/gateway-log-file.{ts,test.ts}` and does not modify settings.local.json (the `Read(//tmp/**)` line is not present on this branch or on main). Resolving as no longer applicable.
